### PR TITLE
add type definitions for heic-convert

### PR DESCRIPTION
**Pull request by treemmett**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/heic-convert/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+     * the JPEG compression quality, between 0 and 1
+     */
+    quality?: number;
+}
+
+interface Convertible {
+    convert(): Promise<ArrayBuffer>;
+}
+
+declare function convert(image: ConversionOptions): Promise<ArrayBuffer>;
```
Comment (peterblazejewicz) [inline types/heic-convert/index.d.ts:25]:
```suggestion
/** @async */
declare function convert(image: ConversionOptions): Promise<ArrayBuffer>;
```
https://jsdoc.app/tags-async.html
(TS declaration file does not support `async` keyword)

File: types/heic-convert/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+declare namespace convert {
+    function all(image: ConversionOptions): Promise<Convertible[]>;
```
Comment (peterblazejewicz) [inline types/heic-convert/index.d.ts:27]:
```suggestion
    /** @async */
    function all(image: ConversionOptions): Promise<Convertible[]>;
```

File: types/heic-convert/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    /**
+     * the HEIC file buffer
+     */
+    buffer: ArrayBufferLike;
+    /**
+     * output format
+     */
+    format: 'JPEG' | 'PNG';
+    /**
+     * the JPEG compression quality, between 0 and 1
```
Comment (peterblazejewicz) [inline types/heic-convert/index.d.ts:16]:
```suggestion
     * the JPEG compression quality, between 0 and 1
     * @default 0.92
```

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
Handle suggestions by @peterblazejewicz

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
There are still comments that are not addressed.

**Comment by treemmett** [comment]
> There are still comments that are not addressed.

@sheetalkamat I apologize, but I'm not seeing any unaddressed comments. Can you point me to the comments you're referring to?

**Comment by sheetalkamat** [comment]
Sorry.. since the conversations were not marked as resolved i missed that you actually fixed it.

**Pull request merged**